pkg/internal/codegen/parse/util.go

```diff
@@ -151,8 +151,22 @@ func HasStatusSubresource(t *types.Type) bool {
 	return false
 }
 
+// HasScaleSubresource returns true if t is an APIResource annotated with
+// +kubebuilder:subresource:scale
+func HasScaleSubresource(t *types.Type) bool {
```

Comment:
wondering if these needs to be *really* exported ? I suspect, no. if not, then lets un-export all of these sort of helpers.

Reply:
For now, the usage is within the `parse` package. It's true, not necessary to export. Will clean up all these util helpers in `parse` package.